components/script/dom/document.rs

```diff
@@ -3491,11 +3495,13 @@ impl Document {
     }
 
     pub fn set_csp_list(&self, csp_list: Option<CspList>) {
-        *self.csp_list.borrow_mut() = csp_list;
+        if let Some(list) = csp_list {
```

Comment:
This is I think a bug, and perhaps the cause of the test failures, because the parser may set this back to none?

Reply:
Thank you Gregory 👍 I was very `fetch` focused when I looked through the spec but there are many places that the spec expects referrer policy to be provided during creation of a new Document or browsing context which I have missed.

As an example: https://html.spec.whatwg.org/multipage/document-sequences.html#creating-browsing-contexts

> Step 16.2: Set document's policy container to a clone of creator's policy container

Is this the intent of `CreatorBrowsingContextInfo`?

The other potential are I saw was https://html.spec.whatwg.org/multipage/document-lifecycle.html#initialise-the-document-object:

> Step 10: Let document be a new [Document](https://html.spec.whatwg.org/multipage/dom.html#document), with ... policy container: navigationParams's [policy container](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigation-params-policy-container)

However `navigationParams` isn't something that looks to be implemented in Servo.